src/vm/excep.cpp

```diff
@@ -7280,8 +7280,17 @@ bool IsIPInMarkedJitHelper(UINT_PTR uControlPc)
 {
     LIMITED_METHOD_CONTRACT;
 
+#ifdef _ARM_
+    // The least bit denoting THUMB might confuse the range check because
+    // PC values have the least bit reset. It causes a bug if PC is at
+    // the first byte of the function range.
+#define CHECK_RANGE(name) \
+    if ((GetEEFuncEntryPoint(name)|0x1) <= (uControlPc|0x1) && (uControlPc|0x1) < (GetEEFuncEntryPoint(name##_End)|0x1)) return true;
```

Comment:
`GetEEFuncEntryPoint` has been showing the address of `name`+1 if the instruction in `name` is Thumb, which made the CHECK_RANGE incorrect if uControlPC is pointing at `name`+0, which often does with JIT_WriteBarrier.

It is right that we only need to set 0x1 of uControlPC; however, can we guarantee that GetEEFuncEntryPoint() is going to be always set 0x1?